theano/gpuarray/tests/test_dnn.py

```diff
@@ -636,75 +639,87 @@ def _test_conv(self, img, kerns, out, img_val, kern_vals, border_mode, conv_mode
                                         [conv_modes[0]])),
                           testcase_func_name=utt.custom_name_func)
     def test_conv(self, algo, border_mode, conv_mode):
+        # Currently only CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_GEMM (algo 'none')
+        # supports dilation > 1.
+        dilations = [(1, 1), (2, 2)] if (algo == "none" and dnn.version() >= 6000) else [(1, 1)]
```

Comment:
I think we should also allow "time_*" and "guess_*" here (which should always select `IMPLICIT_GEMM`, as you say).